src/save/encode.ts

```diff
@@ -29,13 +34,15 @@ const defaultPng: EncodeOptionsPng = { format: 'png' };
  * @returns The encoded image.
  */
 export function encode(
-  image: Image,
-  options: EncodeOptionsPng | EncodeOptionsJpeg = defaultPng,
+  image: Image | Mask,
+  options: EncodeOptionsBmp | EncodeOptionsPng | EncodeOptionsJpeg = defaultPng,
 ): Uint8Array {
   if (options.format === 'png') {
-    return encodePng(image, options.encoderOptions);
+    return encodePng(image as Image, options.encoderOptions);
   } else if (options.format === 'jpg' || options.format === 'jpeg') {
-    return encodeJpeg(image, options.encoderOptions);
+    return encodeJpeg(image as Image, options.encoderOptions);
```

Comment:
I am not sure if it's okay to do it this way. png and jpg do not support Mask, but I didn't find a throw in case a user tries to encode it with these formats. Plus, I believe using `as Image` is not a good practice, but I don't know how to do it otherwise.

Reply:
You added `Mask` to the input types of the `encode` function, so it's up to you to update the other encoders.
I think instead of throwing we should automatically convert the mask to something meaningful inside each encoder, like it's already done for jpeg when the image is not RGBA/8bit.

Reply:
The PNG format supports 1 bit per pixel but I don't know if the encoder does. If it doesn't, we can convert to grey/8bit.

Reply:
https://github.com/image-js/fast-png/blob/ff545ceb792a84ddd5366a0223fbe036e4a9c57d/src/PngEncoder.ts#L152-L159

Seems like encoder doesn't support 1 bit per pixel.